granulate_utils/linux/cgroups/base_cgroup.py

```diff
@@ -5,17 +5,61 @@
 
 import os
 from pathlib import Path
-from typing import List, Tuple
+from typing import List, Mapping, Optional, Tuple
 
 from psutil import NoSuchProcess
 
-CGROUPFS = Path("/sys/fs/cgroup")
-SUBSYSTEMS = {"memory", "cpu"}
-
+from granulate_utils.exceptions import AlreadyInCgroup
+from granulate_utils.linux.mountinfo import iter_mountinfo
+from granulate_utils.linux.ns import resolve_host_root_links
 
-class AlreadyInCgroup(Exception):
-    def __init__(self, subsystem: str, cgroup: str) -> None:
-        super().__init__(f"{subsystem!r} subsytem is already in a predefined cgroup: {cgroup!r}")
+# TODO: This is not always correct. For example, on Amazon Linux 1 this should be /cgroup.
+CGROUPFS = Path("/sys/fs/cgroup")
+SUBSYSTEMS = {
+    "blkio",
+    "cpu",
+    "cpuacct",
+    "cpuset",
+    "devices",
+    "freezer",
+    "hugetlb",
+    "memory",
+    "net_cls",
+    "net_prio",
+    "perf_event",
+    "pids",
+    "rdma",
+}
+
+
+def find_v1_hierarchies() -> Mapping[str, str]:
+    """
+    Finds all the mounted hierarchies for all currently enabled cgroup v1 controllers.
+    :return: A mapping from cgroup subsystem names to their respective hierarchies.
+    """
+    hierarchies = {}
+    for mount in iter_mountinfo(1):
+        # The mount source is always "cgroup" by convention, we don't really have to check it.
+        if mount.mount_source != "cgroup" or mount.filesystem_type != "cgroup":
+            continue
+        controllers = set(mount.super_options) & SUBSYSTEMS
```

Comment:
Btw did u ever see a mountpoint for multiple controllers?

Reply:
```
│ │ ├─/sys/fs/cgroup/cpu,cpuacct      cgroup                 cgroup      rw,nosuid,nodev,noexec,relatime,cpu,cpuacct
```
This is for `cpu`, and `cpuacct`.

Reply:
Here's another:
```
│ │ ├─/sys/fs/cgroup/net_cls,net_prio cgroup                 cgroup      rw,nosuid,nodev,noexec,relatime,net_cls,net_prio
```

Reply:
Cool cool